internal/database/access_requests.go

```diff
@@ -0,0 +1,268 @@
+)
+
+func (s *accessRequestStore) Create(ctx context.Context, accessRequest *types.AccessRequest) (*types.AccessRequest, error) {
+	// We don't allow adding a new request_access with an email address that has already been
+	// verified by another user.
+	exists, _, err := basestore.ScanFirstBool(s.Query(ctx, sqlf.Sprintf("SELECT TRUE WHERE EXISTS (SELECT FROM user_emails WHERE email = %s AND verified_at IS NOT NULL)", accessRequest.Email)))
+	if err != nil {
+		return nil, err
+	}
+	if exists {
+		return nil, errCannotCreateAccessRequest{errorCodeUserWithEmailExists}
+	}
+
+	// We don't allow adding a new request_access with an email address that has already been used
+	exists, _, err = basestore.ScanFirstBool(s.Query(ctx, sqlf.Sprintf("SELECT TRUE WHERE EXISTS (SELECT FROM access_requests WHERE email = %s)", accessRequest.Email)))
+	if err != nil {
+		return nil, err
+	}
+	if exists {
+		return nil, errCannotCreateAccessRequest{errorCodeAccessRequestWithEmailExists}
+	}
+
+	// Continue with creating the new access request.
+	q := sqlf.Sprintf(
+		accessRequestInsertQuery,
+		accessRequest.Name,
+		accessRequest.Email,
+		accessRequest.AdditionalInfo,
+	)
+	var data types.AccessRequest
+
+	if err := s.QueryRow(ctx, q).Scan(&data.ID, &data.CreatedAt, &data.UpdatedAt, &data.Name, &data.Email, &data.Status, &data.AdditionalInfo); err != nil {
+		return nil, errors.Wrap(err, "scanning access_request")
+	}
+
+	return &data, nil
+}
+
+func (s *accessRequestStore) GetByID(ctx context.Context, id int32) (*types.AccessRequest, error) {
+	row := s.QueryRow(ctx, sqlf.Sprintf("SELECT id, created_at, updated_at, name, email, status, additional_info FROM access_requests WHERE id = %s", id))
+	var node types.AccessRequest
+
+	if err := row.Scan(&node.ID, &node.CreatedAt, &node.UpdatedAt, &node.Name, &node.Email, &node.Status, &node.AdditionalInfo); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, &errAccessRequestNotFound{ID: id}
+		}
+		return nil, err
+	}
+
+	return &node, nil
+}
+
+func (s *accessRequestStore) Update(ctx context.Context, accessRequest *types.AccessRequest) (*types.AccessRequest, error) {
+	q := sqlf.Sprintf(accessRequestUpdateQuery, accessRequest.Status, accessRequest.ID)
+	var updated types.AccessRequest
+
+	if err := s.QueryRow(ctx, q).Scan(&updated.ID, &updated.CreatedAt, &updated.UpdatedAt, &updated.Name, &updated.Email, &updated.Status, &updated.AdditionalInfo); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, &errAccessRequestNotFound{ID: accessRequest.ID}
+		}
+		return nil, errors.Wrap(err, "scanning access_request")
+	}
+
+	return &updated, nil
+}
+
+func (s *accessRequestStore) Count(ctx context.Context, opt *AccessRequestsFilterOptions) (int, error) {
+	q := sqlf.Sprintf("SELECT COUNT(*) FROM access_requests WHERE (%s)", sqlf.Join(opt.sqlConditions(), ") AND ("))
+	var count int
+	if err := s.QueryRow(ctx, q).Scan(&count); err != nil {
+		return 0, err
+	}
+	return count, nil
```

Comment:
```suggestion
	return basestore.ScanInt(s.QueryRow(ctx, q))
```

I _think_ this should work